### PR TITLE
feat: cutover charm in actions apiserver

### PR DESCRIPTION
**Pull request by SimonRichardson**
Cut over the actions facade to use the dqlite charm calls. This returns exactly the charm actions, nothing more, nothing less. This should make it more efficient to call this API. There are no other additional things to be concerned about.

## QA steps

The set of QA steps is long-winded, as we've not cut over `UpdateUploadedCharm`.

```sh
$ juju bootstrap lxd test
$ juju add-model default
$ juju deploy ubuntu
$ make repl-install
$ make repl
dqlite > SELECT * FROM namespace_list;
$ DB_NAME="<select the second UUID in the list>" make repl
dqlite > SELECT * FROM charm;
dqlite > INSERT INTO charm_action (charm_uuid, key, description, parallel, execution_group, params) VALUES ('<select the UUID of the charm>', 'test', 'This is a test.', 0, 0, '');
$ juju actions ubuntu
```

## Links

**Jira card:** [JUJU-6549](https://warthogs.atlassian.net/browse/JUJU-6549)

[JUJU-6549]: https://warthogs.atlassian.net/browse/JUJU-6549?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apiserver/facades/client/action/register.go [hunk 1]
```diff
@@ -21,9 +21,18 @@ func Register(registry facade.FacadeRegistry) {
...
 func newActionAPIV7(ctx facade.ModelContext) (*APIv7, error) {
-	api, err := newActionAPI(&stateShim{st: ctx.State()}, ctx.Resources(), ctx.Auth(), ctx.LeadershipReader, ctx.DomainServices().BlockCommand())
+	domainServices := ctx.DomainServices()
+
+	api, err := newActionAPI(
+		&stateShim{st: ctx.State()},
+		ctx.Resources(),
+		ctx.Auth(),
+		ctx.LeadershipReader,
+		domainServices.Application(nil),
```
Comment (jack-w-shaw) [inline apiserver/facades/client/action/register.go:31]: Rebase onto main please, so we can drop this nil

File: apiserver/facades/client/action/action.go [hunk 2]
```diff
@@ -12,21 +12,44 @@ import (
...
+	// Returns [applicationerrors.ApplicationNameNotValid] if the name is not
+	// valid, and [applicationerrors.CharmNotFound] if the charm is not found.
+	GetCharmIDByApplicationName(ctx context.Context, name string) (corecharm.ID, error)
+
+	// GetCharmActions returns the actions for the charm using the charm ID.
+	//
+	// If the charm does not exist, a [applicationerrors.CharmNotFound] error is
+	// returned.
+	GetCharmActions(ctx context.Context, id corecharm.ID) (internalcharm.Actions, error)
+}
```
Comment (jack-w-shaw) [inline apiserver/facades/client/action/action.go:42]: This would make more sense in `interface.go`

**Review by jack-w-shaw — APPROVED** [review]
LGTM

once rebased

**Review by nvinuesa — APPROVED** [review]

**Pull request merged**